src/index.ts

```diff
@@ -2353,20 +2356,26 @@ function getLatestContractVersion({
 			})
 
 		config.verbose && console.log('Contract Versions found:', versions)
+		let _versions = versions
 		// Adjust the filtering logic based on the experimental flag and contract version variables
-		if (!experimental && type === 'normal') {
-			if (!versions.includes(LATEST_STABLE_CONTRACT_VERSION)) {
-				if (versions.length === 0) {
-					versions = [FALLBACK_CONTRACT_VERSION]
+		if (!experimental) {
+			//check if the latest version is an experimental version. On some chains, we've deployed 4.4 but not 4.3. In this case, we want to return 4.4 as latest contract version.
+			if (type === 'normal') {
+				if (versions[0] === LATEST_EXPERIMENTAL_CONTRACT_VERSION) {
+					_versions = versions.filter((version) => version !== LATEST_EXPERIMENTAL_CONTRACT_VERSION)
 				}
-
-				if (LATEST_STABLE_CONTRACT_VERSION !== LATEST_EXPERIMENTAL_CONTRACT_VERSION) {
-					versions = versions.filter((version) => version !== LATEST_EXPERIMENTAL_CONTRACT_VERSION)
+			} else if (type === 'batch') {
```

Comment:
as said, i'd add a TODO: handle router or smth like that, since we also have a router contract that I expect we'll expand over time